contracts/src/ERC20Destination.sol

```diff
@@ -90,9 +88,15 @@ contract ERC20Destination is IERC20Bridge, TeleporterTokenDestination, ERC20 {
 
     /**
      * @dev See {TeleporterTokenDestination-_deposit}
+     *
+     * Note: The amount returned must be the amount credited as a result of the transfer.
+     * For a standard ERC20 implementation such as this contract, that is equal to the full amount given.
+     * For fee/burn on transfer tokens, that amount could be less.
      */
     function _deposit(uint256 amount) internal override returns (uint256) {
-        return SafeERC20TransferFrom.safeTransferFrom(this, amount);
+        _spendAllowance(msg.sender, address(this), amount);
+        _transfer(msg.sender, address(this), amount);
```

Comment:
do we need to check for the actual balance increase, in cases of fee on transfer?

Reply:
My understanding is that, previously, the call to the `SafeERC20TransferFrom.safeTransferFrom` library function was internal, but the implementation of `safeTransferFrom` would make an external call to `approve()` and `transferFrom()` of the given `IERC20` contract. In this case, from the internal library call, we'd essentially be making a call to `address(this).transferFrom()`, which would be an external call back the same contract. 

In regards to checking for the actual balance increase, in this case the `_spendAllowance` and `_transfer` calls are to internal function (i.e. not arbitrary external code), so it is known that it is not a fee on transfer or similar implementation.

Reply:
Good point. I made the internal implementations virtual and made the comment here more explicit about child contracts needing to ensure their own correctness.